src/dex/fluid-dex-lite/fluid-dex-lite.ts

```diff
@@ -603,43 +603,45 @@ export class FluidDexLite
           }
 
           // Get decimals for proper scaling
-          const token0Decimals = Number(unpackedVars.token0Decimals);
-          const token1Decimals = Number(unpackedVars.token1Decimals);
+          const token0Decimals = BigInt(unpackedVars.token0Decimals);
+          const token1Decimals = BigInt(unpackedVars.token1Decimals);
 
           // Calculate geometric mean of liquidity as a proxy for USD value
-          // Convert to 18 decimal standard for calculation
+          // tokenSupply is scaled to 9 decimals for all tokens
           const token0LiquidityNormalized =
-            token0Supply * 10n ** (18n - BigInt(token0Decimals));
+            (token0Supply * 10n ** token0Decimals) / 10n ** 9n;
+
           const token1LiquidityNormalized =
-            token1Supply * 10n ** (18n - BigInt(token1Decimals));
+            (token1Supply * 10n ** token1Decimals) / 10n ** 9n;
```

Comment:
if (token1Decimals < 9) token1LiquidityNormalized = token1Supply / (10n ** (9n - BigInt(token1Decimals)));
else token1LiquidityNormalized = token1Supply * (10n ** (BigInt(token1Decimals) - 9n));

Reply:
suggested version would result in the same output as in the existing code